modules/apps/frontend-js/frontend-js-spa-web/src/main/resources/META-INF/resources/senna/app/App.js

```diff
@@ -1204,12 +1188,14 @@ class App extends EventEmitter {
 					setReferrer(state.referrer);
 				}
 			});
-			const uri = new Uri(state.path);
-			uri.setHostname(globals.window.location.hostname);
-			uri.setPort(globals.window.location.port);
+			const uri = state.path.startsWith('/')
+				? new URL(state.path, globals.window.location.origin)
+				: new URL(state.path);
+			uri.hostname = globals.window.location.hostname;
+			uri.port = globals.window.location.port;
 			const isNavigationScheduled = this.maybeScheduleNavigation_(
 				uri.toString(),
-				{}
+				new Map()
```

Comment:
@wincent, this was causing some `TypeError: event is not Iterable` or something like that error. As it turns out, `{}` is not iterable by default so `...event` somewhere up the chain would throw. 

Not sure what should be the better way to deal with this. Any "empty" iterable would do the job here, just not so sure what we want to do... 🤔

Reply:
Haven't analyzed this deeply but...

The fact that `maybeScheduleNavigation_()` spreads in `...event` into `scheduledNavigationQueue` seems sketchy at best; it seems to be the intent is to "copy" the DOM event, which is usually a pretty manual process and I wouldn't expect spread to do a very good job of it. Maybe it should just be holding a reference to the event, not a copy of it?

Also, the queue itself is an array of events _or_ objects, yet there are places in the code that assume that they really are events (ie. calling `preventDefault()` on them). All seems a bit fragile.

Reply:
Indeed... 😢